source/lib/vagrant-openstack-provider/config_resolver.rb

```diff
@@ -87,9 +85,14 @@ def resolve_volume_boot(env)
         @logger.debug(volume_list)
 
         volume = resolve_volume(config.volume_boot, volume_list, volume_ids)
-        device = volume[:device].nil? ? 'vda' : volume[:device]
 
-        { id: volume[:id], device: device }
+        device = (volume[:device].nil?) ? 'vda' : volume[:device]
+        size = (volume[:size].nil?) ? nil : volume[:size]
+        delete_on_destroy = (volume[:delete_on_destroy].nil?) ? nil : volume[:delete_on_destroy]
+
+        image = resolve_image(env, volume[:image]) unless volume[:image].nil?
+        image_id = (image.nil?) ? nil : image.id
+        { id: volume[:id], image: image_id, device: device, size: size, delete_on_destroy: delete_on_destroy }
```

Comment:
It is confusing to always return the same hash when we want to attach an existing volume or attach a newly created one. It would be better to return a more specific hash. I would replace the last two lines by :

```ruby
if image.nil?
  return { id: volume[:id], device: device }
end
{ image: image_id, device: device, size: size, delete_on_destroy: delete_on_destroy }
```